pathd/path_pcep_lib.c

```diff
@@ -54,28 +66,41 @@ static void free_counter(struct counter *counter);
 
 /* ------------ API Functions ------------ */
 
-int pcep_lib_initialize(void)
+int pcep_lib_initialize(struct frr_pthread *fpt)
 {
 	PCEP_DEBUG("Initializing pceplib");
 
+	/* Register pceplib logging callback */
+	register_logger(pceplib_logging_cb);
+
 	/* Its ok that this object goes out of scope, as it
 	 * wont be stored, and its values will be copied */
 	struct pceplib_infra_config infra = {
+		/* Memory infrastructure */
 		.pceplib_infra_mt = MTYPE_PCEPLIB_INFRA,
 		.pceplib_messages_mt = MTYPE_PCEPLIB_MESSAGES,
-		.mfunc = (pceplib_malloc_func)  qmalloc,
-		.cfunc = (pceplib_calloc_func)  qcalloc,
-		.rfunc = (pceplib_realloc_func) qrealloc,
-		.sfunc = (pceplib_strdup_func)  qstrdup,
-		.ffunc = (pceplib_free_func)    qfree
+		.malloc_func = (pceplib_malloc_func)qmalloc,
+		.calloc_func = (pceplib_calloc_func)qcalloc,
+		.realloc_func = (pceplib_realloc_func)qrealloc,
+		.strdup_func = (pceplib_strdup_func)qstrdup,
+		.free_func = (pceplib_free_func)qfree,
+		/* Timers infrastructure */
+		.external_infra_data = fpt,
+		.timer_create_func = pcep_lib_pceplib_timer_create_cb,
+		.timer_cancel_func = pcep_lib_pceplib_timer_cancel_cb,
+		/*.timer_create_func   = NULL,
```

Comment:
What are these comments for ?

Reply:
I will remove these comments, I had them in for testing purposes.